warehouse/query-core/src/main/java/datawave/query/jexl/functions/IdentityAggregator.java

```diff
@@ -102,11 +104,12 @@ public Key apply(SortedKeyValueIterator<Key,Value> itr, Document doc, AttributeF
             
             Attribute<?> attr = attrs.create(fieldNameValue.first(), fieldNameValue.second(), topKey, true);
             // only keep fields that are index only and pass the attribute filter
-            boolean toKeep = (fieldsToKeep == null || fieldsToKeep.contains(JexlASTHelper.removeGroupingContext(fieldNameValue.first())))
-                            && (filter == null || filter.keep(topKey));
+            boolean toKeep = toKeep(topKey, fieldNameValue);
             attr.setToKeep(toKeep);
-            doc.put(fieldNameValue.first(), attr);
-            key = nextKey;
+            // only add to the document if the filter passes or we are keeping it
```

Comment:
I think this comment is wrong: // only add to the document if the filter passes
the apply is the only gateway to adding to the document.  After adding, the keep flag is set based on the toKeep method.

Reply:
We may consider removing the addToDoc method and simply use the filter here to avoid confusion.